sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CompactibleFileStreamLog.scala

```diff
@@ -388,7 +388,7 @@ object CompactibleFileStreamLog {
     } else if (defaultInterval < (latestCompactBatchId + 1) / 2) {
       // Find the first divisor >= default compact interval
       def properDivisors(min: Int, n: Int) =
-        (min to n/2).view.filter(i => n % i == 0).toSeq :+ n
+        (min to n / 2).to(LazyList).filter(i => n % i == 0) :+ n
```

Comment:
@yaooqinn 
https://github.com/apache/spark/pull/15852/files#diff-ed29e1893fa3939724b333684e1f6037358b3d7edce1e3663afdda8d99002fed

![image](https://github.com/apache/spark/assets/1475305/03e8f4dc-cada-48de-8d8d-206ad1dc6d83)

seems that the initial intention here was also to use a lazy view